File: addDocuments.py
import sys
import typesense
import unicodedata

fileName = sys.argv[1]

client = typesense.Client({
  'nodes': [{
    'host': 'localhost',  # For Typesense Cloud use xxx.a1.typesense.net
    'port': '8108',       # For Typesense Cloud use 443
    'protocol': 'http'    # For Typesense Cloud use https
  }],
  'api_key': 'xyz',
  'connection_timeout_seconds': 2
})

# print(fileName)
data = open(fileName,'r').read()
# jsonl_file = jsonl_file.encode('latin-1', 'replace')


# print(data[5861456:5861476])
# print(str(unicodedata.normalize('NFKD', jsonl_file[5814010:5814020]).encode('ascii', 'ignore')))
# data = unicodedata.normalize('NFKD', jsonl_file).encode('ascii', 'ignore').decode("utf-8")

return_data = client.collections['blogs'].documents.import_(data)
print('Documents length: ', len(data.split('\n'))-1)
print('Sucessfully added: ', return_data.count('true'))

File: addDocuments.js
const Typesense = require('typesense');
const fs = require('fs');
module.exports = (async() => {
    const typesense = new Typesense.Client({
        nodes: [{
            host: '172.105.148.128',
            port: '8108',
            protocol: 'http',
        }, ],
        apiKey: 'xyz',
    });

    var myArgs = process.argv.slice(2);

    console.log('Adding records: ');

    try {
        const docs = fs.readFile(myArgs[0], 'utf8', (err, data) => {
            // console.log(err);
        });
        const returnData = await typesense
            .collections('blogs')
            .documents()
            .import(docs, { batch_size: 100 });
        console.log(returnData);
        console.log('Done indexing.');

        const failedItems = returnData.filter(item => item.success === false);
        if (failedItems.length > 0) {
            throw new Error(
                `Error indexing items ${JSON.stringify(failedItems, null, 2)}`
            );
        }

        return returnData;
    } catch (error) {
        console.log(error);
    }
})();